### PR TITLE
PLAT-707: test deduplicate deactivation

### PR DESCRIPTION
**Pull request by MariaVasilenko**
case https://insolar.testrail.io/index.php?/cases/view/5507

### CODE REVIEW

File: ledger-core/virtual/integration/deactivate_test.go [hunk 1]
```diff
@@ -746,3 +746,129 @@ func TestVirtual_CallMethod_After_Deactivation(t *testing.T) {
...
+		},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestResult,
+			},
+		)
+		runnerMock.AddExecutionClassify(outgoingDeactivate.String(), isolation, nil)
+
+		// Constructor mock
+		result := requestresult.New([]byte("new"), objectRef)
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/deactivate_test.go:800]: add sideEffect for constructor   
```
result := requestresult.New([]byte("new"), objectRef)
result.SetActivate(reference.Global{}, class, []byte("initial state"))
```

File: ledger-core/virtual/integration/deactivate_test.go [hunk 2]
```diff
@@ -746,3 +746,129 @@ func TestVirtual_CallMethod_After_Deactivation(t *testing.T) {
...
+
+	requests := []payload.VCallRequest{pl, deactivateRequest, pl, deactivateRequest}
+	i := 1
+	for _, r := range requests {
+		await := server.Journal.WaitStopOf(&execute.SMExecute{}, i)
+		server.SendPayload(ctx, &r)
+		commonTestUtils.WaitSignalsTimed(t, 10*time.Second, await)
+		commonTestUtils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
+		i++
+	}
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/deactivate_test.go:866]:
```suggestion
	for _, r := range requests {
		await := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
		server.SendPayload(ctx, &r)
		commonTestUtils.WaitSignalsTimed(t, 10*time.Second, await)
	}

	commonTestUtils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
```

File: ledger-core/virtual/integration/deactivate_test.go [hunk 3]
```diff
@@ -746,3 +746,129 @@ func TestVirtual_CallMethod_After_Deactivation(t *testing.T) {
...
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestResult,
+			},
+		)
+		runnerMock.AddExecutionClassify(outgoingDeactivate.String(), isolation, nil)
+
+		// Constructor mock
+		result := requestresult.New([]byte("new"), objectRef)
+		constructorMock := runnerMock.AddExecutionMock(outgoing.String())
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/deactivate_test.go:801]:
```suggestion
		result := requestresult.New([]byte("new"), outgoing)
		result.SetActivate(reference.Global{}, class, []byte("state"))
```

File: ledger-core/virtual/integration/deactivate_test.go [hunk 4]
```diff
@@ -746,3 +746,129 @@ func TestVirtual_CallMethod_After_Deactivation(t *testing.T) {
...
+	runnerMock := logicless.NewServiceMock(ctx, t, nil)
+	server.ReplaceRunner(runnerMock)
+	server.Init(ctx)
+	server.IncrementPulseAndWaitIdle(ctx)
+
+	var (
+		class              = gen.UniqueGlobalRef()
+		objectRef          = reference.NewSelf(server.RandomLocalWithPulse())
+		outgoing           = server.BuildRandomOutgoingWithPulse()
+		outgoingDeactivate = server.BuildRandomOutgoingWithPulse()
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/deactivate_test.go:768]:
```suggestion
		outgoing           = server.BuildRandomOutgoingWithPulse()
		objectRef          = reference.NewSelf(outgoing.GetLocal())
		outgoingDeactivate = server.BuildRandomOutgoingWithPulse()
```

File: ledger-core/virtual/integration/deactivate_test.go (hunk 2 above)
Comment (MariaVasilenko) [reply ledger-core/virtual/integration/deactivate_test.go]: thanks

File: ledger-core/virtual/integration/deactivate_test.go (hunk 4 above)
Comment (MariaVasilenko) [reply ledger-core/virtual/integration/deactivate_test.go]: fixed

File: ledger-core/virtual/integration/deactivate_test.go (hunk 3 above)
Comment (MariaVasilenko) [reply ledger-core/virtual/integration/deactivate_test.go]: fixed

**Pull request merged**